### PR TITLE
OCM Add cluter-backup-operator to PROW

### PR DESCRIPTION
**Pull request by jnpacker**
Signed-off-by: Joshua Packer <jpacker@redhat.com>

* Add the cluster-backup-operator repository to PROW.
* Creates image cluster-backup-controller.

### CODE REVIEW

File: ci-operator/config/open-cluster-management/cluster-backup-operator/open-cluster-management-cluster-backup-operator-main.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+base_images:
+  open-cluster-management_builder_go1.16-linux:
+    name: builder
+    namespace: open-cluster-management
+    tag: go1.16-linux
+  repo-brancher:
+    name: repo-brancher
+    namespace: ci
+    tag: latest
```
Comment (ray-harris) [inline ci-operator/config/open-cluster-management/cluster-backup-operator/open-cluster-management-cluster-backup-operator-main.yaml:9]: Fast forward process has changed. These lines aren't needed. See comment below for more info.

File: ci-operator/config/open-cluster-management/cluster-backup-operator/open-cluster-management-cluster-backup-operator-main.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        --token-path=/etc/github/token \
+        --current-promotion-namespace=open-cluster-management \
+        --current-release=2.4 \
+        --future-release=2.4
+  container:
+    from: repo-brancher
+  postsubmit: true
+  secrets:
+  - mount_path: /etc/github
+    name: acm-cicd-github
```
Comment (ray-harris) [inline ci-operator/config/open-cluster-management/cluster-backup-operator/open-cluster-management-cluster-backup-operator-main.yaml:77]: Fast forward process has changed, but docs have not yet been updated. Use this stanza instead:

```
- as: fast-forward
  postsubmit: true
  steps:
    env:
      DESTINATION_BRANCH: release-2.4
    workflow: ocm-ci-fastforward
```

**Review by ray-harris — CHANGES_REQUESTED** [review]
You also need to add an OWNERS file in this directory:

```
ci-operator/jobs/open-cluster-management/cluster-backup-operator
```

**Comment by ray-harris** [comment]
The rehearsal jobs to build the images should be passing. Looking at them, it looks like an issue with Prow itself. I'll have to check with the Prow admins about it.

**Comment by jnpacker** [comment]
I applied the fixes your recommended. I'm still waiting for CICD to create the cluster-backup-controller repo in quay as well.

**Comment by ray-harris** [comment]
@jnpacker In the config file for the release-2.4 branch, you're referencing `Dockerfile.prow`, but that doesn't exist in the `cluster-backup-operator` repo.

**Comment by jnpacker** [comment]
yeah I had a pull to fix that on THU but everyone had headed out. Will be fixing today. 😄

**Pull request closed without merging**